src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt

```diff
@@ -62,3 +74,60 @@ private class RsLiteralFileReferenceSet(
         }
     }
 }
+
+private val pathValueLiteral: PsiElementPattern.Capture<RsLitExpr> = psiElement<RsLitExpr>()
+    .with("onPathLiteral") { expr -> expr.kind is RsLiteralKind.String && isPathLitExpr(expr) }
+
+private fun isPathLitExpr(expr: RsLitExpr): Boolean {
+    val (function, arguments) = getFunctionAndArguments(expr) ?: return false
+    val owner = ((function.owner as? RsAbstractableOwner.Impl)?.impl?.typeReference?.type as? TyAdt)?.item
+    val (implLookup, knownItems) = expr.implLookupAndKnownItems
+    val isCallExpr = arguments.parent is RsCallExpr
+
+    return when {
+        // Path::new(<literal>)
+        owner == knownItems.Path && function.name == "new" -> true
```

Comment:
I added the check for inherent impl, but it doesn't work in this case, because the function is resolved to `Path::new` instead of `My::new`. There is a resolve bug probably.

Reply:
Hm, indeed. But let's still add this check. When name resolution will be fixed, the check will start to work

Reply:
Ok, I'll create a new issue for the name resolution.

Reply:
Probably, we already have such issue but I'm not sure

Reply:
I couldn't any similar issue, so I created a new one: https://github.com/intellij-rust/intellij-rust/issues/5483.